src/Uno.UWP/Media/Playback/MediaPlayer.Android.cs

```diff
@@ -134,21 +135,23 @@ protected virtual void InitializeSource()
 
 				PlaybackSession.PlaybackState = MediaPlaybackState.Opening;
 
-				if (Source is MediaPlaybackList)
+				Uri uri;
+				switch (Source)
 				{
-					_playlistItems = new List<Uri>();
-
-					var playlist = Source as MediaPlaybackList;
-					foreach (var mediaItem in playlist.Items)
-					{
-						_playlistItems.Add(mediaItem.Source.Uri);
-					}
+					case MediaPlaybackList playlist when playlist.Items.Count > 0:
+						SetPlaylistItems(playlist);
+						uri = _playlistItems[0];
+						break;
+					case MediaPlaybackItem item:
+						uri = item.Source.Uri;
+						break;
+					case MediaSource source:
+						uri = source.Uri;
+						break;
+					default:
+						throw new InvalidOperationException("Unsupported media source type");
```

Comment:
Is this going to throw if it receives an empty `MediaPlaybackList`?

Reply:
Yes it would throw, it would be caught and `OnMediaFailed` would be called. Perhaps it needs a better error message?

In the previous version it would have thrown an `InvalidCastException` because it would assume `Source` is of type `MediaSource` if the list was empty

Reply:
> Perhaps it needs a better error message?

Really it should be consistent with whatever Windows does, and also iOS for that matter which seems to permit an empty list.

As you note though it would've failed before, so it's fine for this PR - just create an issue noting that it should be addressed. (Another good Hacktoberfest issue 😛 )